careless-gazelles/Bangazon-Site
# Oo complete order

## Status
Choose:
**READY**

## Description
Complete ticket #2. User can view list of all orders and add PaymentType to order that isn't completed. Once PaymentType is added, DateCompleted is also added.

The logic is setup to where you can only complete an order that has a NULL PaymentType. If you have no open orders, you'll need to add one directly to your database. Orders --> 'Show Table Data' --> type in required info (leave DateCompleted blank). Add some products to it by repeating the process in the Products and OrderProducts tables.

Had to add a ViewModel and refactor some of the Order views.

There were a couple bugs with editing Products and Orders so I squashed 'em. Real flat.

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

/Orders
'Details'
'Complete Order' -- should only show for incomplete orders

```
git fetch --all
git branch -avv
git checkout oo-completeOrder
git pull origin oo-completeOrder
```
Then Open Visual Studio

```
dotnet restore
dotnet run
```

## Impacted Areas in Application
List general components of the application that this PR will affect:
Ticket #2 
Ticket #16 
OrderViewModels
Order.Create
Order.Edit
Order.Delete
Order.Confirmation
Order.Details
OrdersController
ProductsController
